include/envoy/server/configuration.h

```diff
@@ -89,6 +89,11 @@ class Main {
    */
   virtual std::chrono::milliseconds statsFlushInterval() const PURE;
 
+  /**
+   * @return bool indicator to flush stats on-demand via the admin interface instead of on a timer.
+   */
+  virtual bool statsFlushOnAdmin() const PURE;
```

Comment:
Instead of polluting the main interfaces with various stats config, can we make a new interface for `StatsConfig` and just pass that around? Same below.

Reply:
This makes sense to me. Should we reflect the same in the v4 api?

Reply:
I don't think I would bother with an API change here, just internal code change.

Reply:
sounds good